src/assets/definitions.tsx

```diff
@@ -1,48 +1,58 @@
-export interface ApplicantCard extends SearchResponse {
-  query: {query: Query}
+// types
+export type Profile = {
+	id?: number
+	first_name?: string
+	last_name?: string
+	username: string
+	email?: string
+	bio: string
 }
 
-export interface ApplicantProfile {
-	username:string,
-	bio:string,
-	skills:Array<Attribute>,
-	values:Array<Attribute> 
+export type Attribute = {
+	attribute?: string
+	id?: number
 }
 
-export interface SearchResponse {
-  id: number
-  username: string
-  bio: string
-  skills: Array<Attribute>
-  values: Array<Attribute>
+export type ApiAttribute = {
+	attribute: string
+	id: number
 }
 
-export interface Applicant {
-  id: number
-	username: string
-	email: string
-  bio: string
-  skills: Array<Attribute>
-  values: Array<Attribute>
+export type Query = {
+	skills: Array<{ attribute: string, checked: boolean }>
+	values: Array<{ attribute: string, checked: boolean }>
 }
 
-export type AttributeList = {
-	skills: Array<Attribute>
-	values: Array<Attribute>
+export type EmployerMessage = {
+	employer_name: string
+	employer_email: string
+	body: string
 }
 
+export type HeaderProps = {
+	auth: number | undefined
+}
 
-// this is the attribute type we will use
-//DO NOT DELETE 
-export type Attribute = {
-	attribute?: string 
-	id?: number
+// interfaces
+export interface ApplicantCard extends ApplicantInfo {
+	query: { query: Query }
+}
+
+export interface NewApplicant extends Profile {
```

Comment:
Very glad we did this! I think `extends` and parameters are going to be the next big things to up our typescript game!